### ISSUE TITLE
VirtualBox 5.0 replaced with 4.3

### ISSUE THREAD

**Issue by mihkel** [kitematic/kitematic#808]
Hi, is there a reason why Kitematic ignored my previously installed VBox 5 and replaced it with 4.3?

**Comment by TeckniX** [comment]
@mihkel VirtualBox 5 was just recently released (7/9/2015) - We check for the VBox version that Kitematic has been QA'd with and confirmed to be stable.

Any help in QA with VBox 5 would be appreciated. If you're familiar with the build process from source of Kitematic, I can point you in the right direction to have Kitematic use VirtualBox 5?

**Comment by pandrew** [comment]
machine is not working with v5, yet.

**Comment by FrenchBen** [comment]
thanks @pandrew for chiming in. 
Adding the docker-machine issue tracker:
https://github.com/docker/machine/issues/1498

**Comment by ChrisCinelli** [comment]
Assuming that the changing Virtualbox version is ok, it is NOT ok.

**Comment by FrenchBen** [comment]
@ChrisCinelli Nothing was assumed, 4.x was the confirmed working version of VBox and 5.0 was just release, which caused the overwrite. 
A patch will be created to include support for 5.x and notify the user if a previous version is being over-written.

**Comment by lutnos** [comment]
Thanks

To be clear, by "a previous version is being overwritten" - I hope that you might mean "the installed version is being overwritten, (whether a previous version or future one)".

How about some way for a user to allow Kinematic to continue even if the version is not the tied one?

Just trying to help

**Comment by openube** [comment]
To second lutnos above, how about allowing Kinematic to continue if a virtualbox is installed unless if the detected version is specific to a known bug.

**Comment by doxxx** [comment]
Just tried this with v0.7.5 and it's working for me.

**Comment by hrathod** [comment]
Just tried with VirtualBox 5 and Kitematic v0.7.5 and received the following error during setup.  Stacktrace from Devtools.  `VBoxManage list vms` returns the following:
```
"Debian8" {311f6486-66e1-41a0-90f6-000ca0c7d563}
"Centos7" {687617d8-d379-4776-8801-521091cf5cc3}
```
Stacktrace:
```
Error: /Users/<redacted>/Downloads/Kitematic (Beta).app/Contents/Resources/resources/docker-machine -D create -d virtualbox --virtualbox-boot2docker-url /Users/<redacted>/Downloads/Kitematic (Beta).app/Contents/Resources/resources/boot2docker.iso --virtualbox-memory 2048 dev returned non zero exit code. Stderr: Error creating machine: Machine dev already exists
You will want to check the provider to make sure the machine and associated resources were properly removed.

You will want to check the provider to make sure the machine and associated resources were properly removed.

    at /Users/<redacted>/Downloads/Kitematic (Beta).app/Contents/Resources/app.asar/utils/Util.js:52:18
    at ChildProcess.<anonymous> (/Users/<redacted>/Downloads/Kitematic (Beta).app/Contents/Resources/app.asar/node_modules/exec/exec.js:91:5)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:169:7)
    at maybeClose (child_process.js:996:16)
    at Socket.<anonymous> (child_process.js:1165:11)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:166:7)
    at Pipe._onclose (net.js:464:12)
```

Further, running `./docker-machine ls` from the Contents/Resources/resources directory results in the following error:
```
./docker-machine ls    
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x1697e1]

goroutine 1 [running]:
github.com/docker/machine/libmachine.(*Host).LoadConfig(0x208c82000, 0x0, 0x0)
	/go/src/github.com/docker/machine/libmachine/host.go:287 +0x3d1
github.com/docker/machine/libmachine.Filestore.loadHost(0x208c8a440, 0x1e, 0x208c11a40, 0x2b, 0x208c11d40, 0x2f, 0x208c62298, 0x3, 0x5566a0, 0x0, ...)
	/go/src/github.com/docker/machine/libmachine/filestore.go:34 +0x287
github.com/docker/machine/libmachine.Filestore.Get(0x208c8a440, 0x1e, 0x208c11a40, 0x2b, 0x208c11d40, 0x2f, 0x208c62298, 0x3, 0x0, 0x0, ...)
	/go/src/github.com/docker/machine/libmachine/filestore.go:112 +0x65
github.com/docker/machine/libmachine.Filestore.List(0x208c8a440, 0x1e, 0x208c11a40, 0x2b, 0x208c11d40, 0x2f, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/docker/machine/libmachine/filestore.go:88 +0x2c6
github.com/docker/machine/libmachine.(*Filestore).List(0x208c620f0, 0x0, 0x0, 0x0, 0x0, 0x0)
	<autogenerated>:8 +0xd3
github.com/docker/machine/libmachine.(*Provider).List(0x208c69eb0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/machine/libmachine/provider.go:75 +0x6d
github.com/docker/machine/commands.cmdLs(0x208c886c0)
	/go/src/github.com/docker/machine/commands/ls.go:30 +0x1d7
github.com/codegangsta/cli.Command.Run(0x772210, 0x2, 0x0, 0x0, 0x788310, 0xd, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/docker/machine/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:113 +0x1038
github.com/codegangsta/cli.(*App).Run(0x208bf2270, 0x208bde000, 0x2, 0x2, 0x0, 0x0)
	/go/src/github.com/docker/machine/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:156 +0xcf7
main.main()
	/go/src/github.com/docker/machine/main.go:114 +0x9bc

goroutine 5 [syscall]:
os/signal.loop()
	/usr/src/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/src/go/src/os/signal/signal_unix.go:27 +0x35
```

**Comment by FrenchBen** [comment]
@hrathod Thank you for the feedback - There's a docker-machine  RC that allows VBox 5 support, see:
https://github.com/docker/machine/pull/1496

The one currently shipped with Kitematic does not and will need to be updated once the RC is moved to stable.

**Comment by jeffdm** [comment]
Fixed, thanks!

**Comment by ldmbouge** [comment]
I still had the issue on August 26. Worse, I was happily using boot2docker and succumbed to the prompt to upgrade and start using 'docker-machine' as boot2docker was now deprecated. Upgrading did break the environment. I was worried I'd have to rebuild the whole thing until I found this page. It would be really nice to have a working version (right out of the box) as I will recommend to students to grab docker for my course this coming fall...

**Comment by FrenchBen** [comment]
@ldmbouge What VirtualBox version are you using? There have been reports of issues with 5.0.2 - The test build of 5.0.3 seems to work better, but hasn't been officially released just yet.

**Comment by ldmbouge** [comment]
It was DockerToolbox-1.8.1b.pkg  downloaded from:https://www.docker.com/toolbox  and I grabbed the Mac Version. i.e., this link:

https://github.com/docker/toolbox/releases/download/v1.8.1b/DockerToolbox-1.8.1b.pkg


I’m running OSX 10.10.5

The installer upgraded virtualbox automatically (I used to have 4.3.x) and failed as soon as I asked (via the command-line) to upgrade my docker machines.

I’d have to rerun the installer to check the version of VirtualBox it pulled. But it was a 5.0.x

I can email the installer I grabbed, but since the link is above, it should be the same thing.

What I’m concerned is that, next week, I’ll tell 63 students to install docker in the same way to setup their development environment (grab an image sitting on docker hub) and if their first experience is a flop, I’m going to catch some flack. (I’m in a CS department and I’m teaching a compiler course… putting all the llvm stuff in a container ;-))

—
  Laurent
